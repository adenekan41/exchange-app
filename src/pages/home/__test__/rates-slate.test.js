/* eslint-disable no-undef */
/* -------------------------- External Dependencies ------------------------- */
import React from 'react';
import ReactDOM from 'react-dom';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Router } from 'react-router-dom';

/* -------------------------- Internal Dependencies ------------------------- */
import history from 'utils/history';
import RateSlate from '../components/rates-slate';

const mockStore = configureStore([]);

describe('RateSlate Component', () => {
	afterEach(cleanup);
	let store;
	beforeAll(() => {
		store = mockStore({
			exchange: {
				currentRate: 100,
				from: 'USD',
				to: 'EUR',
			},
		});
		store.dispatch = jest.fn();
	});
	it('RateSlate Renders Without Crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(
			<Provider store={store}>
				<Router history={history}>
					<RateSlate />
				</Router>
			</Provider>,
			div
		);
		ReactDOM.unmountComponentAtNode(div);
	});
	it('RateSlate matches snapshot', () => {
		const tree = renderer
			.create(
				<Provider store={store}>
					<Router history={history}>
						<RateSlate />
					</Router>
				</Provider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('RateSlate matches snapshot without actions', () => {
		const tree = renderer
			.create(
				<Provider store={store}>
					<Router history={history}>
						<RateSlate />
					</Router>
				</Provider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
