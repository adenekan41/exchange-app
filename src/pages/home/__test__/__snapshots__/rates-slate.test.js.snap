// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RateSlate Component RateSlate matches snapshot 1`] = `
<h4
  className="sc-AxjAm jCIwtX"
>
  1
  $
   = 
  100.000
  €
</h4>
`;

exports[`RateSlate Component RateSlate matches snapshot without actions 1`] = `
<h4
  className="sc-AxjAm jCIwtX"
>
  1
  $
   = 
  100.000
  €
</h4>
`;
